HopsanGUI/HcomHandler.cpp

```diff
@@ -6006,124 +6008,51 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
-    {
-        QString argStr = expr.mid(5, expr.size()-6).trimmed();
+    //Simple mathematical vector functions with no arguments
+    FuncMap_t funcMap;
+    funcMap.insert("sin", qMakePair(static_cast<ScalarMathFunction_t>(sin),&VectorVariable::sinOfData));
+    funcMap.insert("cos", qMakePair(static_cast<ScalarMathFunction_t>(cos),&VectorVariable::cosOfData));
+    funcMap.insert("tan", qMakePair(static_cast<ScalarMathFunction_t>(tan),&VectorVariable::tanOfData));
+    funcMap.insert("asin", qMakePair(static_cast<ScalarMathFunction_t>(asin),&VectorVariable::asinOfData));
+    funcMap.insert("acos", qMakePair(static_cast<ScalarMathFunction_t>(acos),&VectorVariable::acosOfData));
+    funcMap.insert("atan", qMakePair(static_cast<ScalarMathFunction_t>(atan),&VectorVariable::atanOfData));
+    funcMap.insert("log", qMakePair(static_cast<ScalarMathFunction_t>(log),&VectorVariable::logOfData));
+    funcMap.insert("exp", qMakePair(static_cast<ScalarMathFunction_t>(exp),&VectorVariable::expOfData));
+    funcMap.insert("sqrt", qMakePair(static_cast<ScalarMathFunction_t>(sqrt),&VectorVariable::sqrtOfData));
+    funcMap.insert("round", qMakePair(static_cast<ScalarMathFunction_t>(round),&VectorVariable::roundOfData));
+    funcMap.insert("floor", qMakePair(static_cast<ScalarMathFunction_t>(floor),&VectorVariable::floorOfData));
+    funcMap.insert("ceil", qMakePair(static_cast<ScalarMathFunction_t>(ceil),&VectorVariable::ceilOfData));
+    funcMap.insert("abs", qMakePair(static_cast<ScalarMathFunction_t>(fabs),&VectorVariable::absOfData));
+    if(desiredType != Scalar && funcMap.contains(getFunctionName(expr))) {
+        QString funcName = getFunctionName(expr);
+        QString argStr = expr.mid(funcName.size()+1, expr.size()-funcName.size()-2).trimmed();
+        ScalarMathFunction_t scalarFunc = funcMap.value(funcName).first;
+        VectorMathFunction_t vectorFunc = funcMap.value(funcName).second;
         SharedVectorVariableT pData = getLogVariable(argStr);
-        if(!pData)
-        {
+        if(!pData) {
             evaluateExpression(argStr);
 
-            if(mAnsType == HcomHandler::DataVector)
-            {
+            if(mAnsType == HcomHandler::DataVector) {
                 pData = mAnsVector;
             }
-            else if (mAnsType == HcomHandler::Scalar)
-            {
-                mAnsScalar = qCeil(mAnsScalar);
+            else if (mAnsType == HcomHandler::Scalar) {
+                mAnsScalar = scalarFunc(mAnsScalar);
                 return;
             }
         }
 
         if(pData)
         {
             mAnsType = HcomHandler::DataVector;
-            mAnsVector = pLogDataHandler->createOrphanVariable(QString("Ceil%1").arg(pData->getSmartName()), pData->getVariableType());
-            mAnsVector->assignFrom(pData->getSharedTimeOrFrequencyVector(), pData->ceilOfData());
+            LogDataHandler2 *pLogDataHandler = mpModel->getViewContainerObject()->getLogDataHandler().data();
```

Comment:
Is this line needed ? Was it not called on line 6008

Reply:
No, you are right. I will remove it.